### PR TITLE
Use more descriptive struct field names

### PR DESCRIPTION
**Pull request by michaelharo**

### CODE REVIEW

**Comment by bogosj** [comment]
@uhthomas and @andig might have some comments on this. Unclear if their services use these values. In general I'm in favor of the change, but I'm not clear on what deviating from the JSON fields would do to existing consumers.

**Review by bogosj — APPROVED** [review]

**Comment by uhthomas** [comment]
I'm a fan of making the API a little nicer. "FrontTrunk" is much clearer than "Ft", but I am wary of changing the `interface{}` values. The Tesla JSON API uses a lot of `null` and knowing the difference between unset, zero and actual will become challenging otherwise. Don't get me wrong, using an `interface{}` is not correct and it should be changed.

**Comment by michaelharo** [comment]
> I'm a fan of making the API a little nicer. "FrontTrunk" is much clearer than "Ft", but I am wary of changing the `interface{}` values. The Tesla JSON API uses a lot of `null` and knowing the difference between unset, zero and actual will become challenging otherwise. Don't get me wrong, using an `interface{}` is not correct and it should be changed.

I haven't checked if this works, but would making the fields pointers resolve the concern about null vs zero value?

File: states.go [hunk 1]
```diff
@@ -219,6 +219,9 @@ type timeSecs struct {
 }
 
 func (t *timeSecs) UnmarshalJSON(b []byte) error {
+	if string(b) == "null" {
```
Comment (andig) [inline states.go:222]: Not sure about these. Might as well make them pointers to time if they're optional?

**Comment by andig** [comment]
+1 for the naming changes. Compatibility is broken anyway, so why not improve usability as well?

**Comment by bogosj** [comment]
Is there anything left to do here before merging?

This plus the login PR should make #16 doable?

**Comment by andig** [comment]
+1

File: states.go (hunk 1 above)
Comment (uhthomas) [reply states.go:222]: Did we ever get a consensus on this?

Comment (andig) [reply states.go:222]: Current proposal will always have the struct field as zero time instead of nil pointer. Probably more „compatible“ with jsgoecke.

**Comment by michaelharo** [comment]
I'm unclear on the direction to take here.  I'd prefer to use nicer types and nicer names.  Should I flip every field over to a pointer so it's possible to check unset vs zero value?

**Comment by bogosj** [comment]
Suggestion: we defer these decisions for now and tag v1.0 as close to compatibility with jsgoecke as possible (plus the auth bits). We can then figure out what cleanup breaking changes should go into v1.1.

**Comment by michaelharo** [comment]
How about we split this change into 2 parts...  The first part, renaming some fields, goes into 1.0.  We can then defer the decision about types and zero vs nil.

**Comment by bogosj** [comment]
I think this is the last thing we are waiting on to tag v1.0?

**Comment by andig** [comment]
> I think this is the last thing we are waiting on to tag v1.0?

We need #43 too :O. 

> How about we split this change into 2 parts... The first part, renaming some fields, goes into 1.0. We can then defer the decision about types and zero vs nil.

Looking at this again, I would suggest to change all field names/types before 1.0 and leave the nil discussion until later. These changes would be easy to pick ip without need to change the logic.

**Comment by michaelharo** [comment]
sorry about the delay.  The time type change has been reverted.

**Comment by andig** [comment]
Lgtm

**Comment by bogosj** [comment]
LGTM. Merge as you see fit.

**Pull request merged**